lightning_transformers/task/nlp/question_answering/datasets/squad/metric.py

```diff
@@ -0,0 +1,30 @@
+import torch
+from datasets import load_metric
+from torchmetrics import Metric
+
+
+class SquadMetric(Metric):
+
+    def __init__(self, postprocess_func, example_id_strings):
+        super().__init__(compute_on_step=False)
+        self.metric = load_metric("squad")
+        self.postprocess_func = postprocess_func
+        self.example_id_strings = example_id_strings
+        self.add_state("start_logits", [])
+        self.add_state("end_logits", [])
+        self.add_state("example_ids", [])
```

Comment:
I'm not sure if we need to store `example_ids`. Currently, the code assumes that the outputs are in the same order as the inputs. If that assumption can break (e.g. if update ordering is not preserved during distributed training), then we may need to sort by `example_ids` before passing the predictions to `postprocess_func` in `compute`.

Reply:
Great point, I'll need to read through the implementation, but I think this was also the case when I first looked at the squad v2 metric!